### ISSUE TITLE
[BUG] Use LoadBalancer Type Service for non-HTTP traffic

### ISSUE THREAD

**Issue by tgdfool2** [kubeovn/kube-ovn#4640]
### Kube-OVN Version

v1.12.18

### Kubernetes Version

Client Version: v1.28.10
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.10

### Operation-system/Kernel Version

Ubuntu 22.04.4 LTS
5.15.0-1063

### Description

Dear Community,

We are currently using a LoadBalancer Type Service (https://kube-ovn.readthedocs.io/zh-cn/latest/en/guide/loadbalancer-service/) to configure the public (LoadBalancer) IP address for an Ingress Nginx Controller.

Everything is working great for HTTP(S)-based traffic but now, we need to extend our setup to expose some non-HTTP services as well (e.g., SSH) to customers. We decided to experiment with the following setup: https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services/, which is working well on "non kube-ovn enabled clusters". On our kube-ovn cluster, we are observing the following behaviour:

_(10.224.1.112 is the public IP address being served by the Ingress Nginx Controller)_
```
# tcpdump -i eth0 port 30001
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on eth0, link-type EN10MB (Ethernet), snapshot length 262144 bytes
12:54:44.717376 IP 10.112.8.19.57130 > 10.224.1.112.30001: Flags [SEW], seq 3231610703, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
12:54:47.715840 IP 10.112.8.19.57130 > 10.224.1.112.30001: Flags [SEW], seq 3231610703, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
12:54:53.716546 IP 10.112.8.19.57130 > 10.224.1.112.30001: Flags [S], seq 3231610703, win 8192, options [mss 1460,nop,nop,sackOK], length 0
```

Basically, requests are reaching the interface, but nobody is answering back with a SYN/ACK.

On the ingress-nginx Pod itself, a tcpdump shows that no traffic is coming to the container.

Questions:
* Is this setup supposed to work with kube-ovn and LoadBalancer Type Service?
* If so, then has something to be configured to make it to work?
* How can we further troubleshoot this issue?

Thank you very much in advance for your help!

### Steps To Reproduce

See Description.

### Current Behavior

Non-HTTP traffic not working through LoadBalancer Type Service

### Expected Behavior

Non-HTTP traffic working through LoadBalancer Type Service

**Comment by oilbeater** [comment]
Looks like a bug in multiple-port service load balancer can you try `kubectl ko nbctl lr-nat-list {VPC NAME}` to check if there is the right mapping?

**Comment by tgdfool2** [comment]
This setup has not been done in a specific VPC; it is meant to be used for the default Ingress Nginx controller:
```
# k -n ingress-nginx get pods
NAME                                              READY   STATUS    RESTARTS   AGE
ingress-nginx-controller-57ccb8d5ff-85mrk         1/1     Running   0          2d23h
lb-svc-ingress-nginx-controller-7984dffd6-6b8jg   1/1     Running   0          5d21h
```

**Comment by tgdfool2** [comment]
Empty output:
```
# kubectl ko nbctl lr-nat-list ovn-cluster
#
```

**Comment by tgdfool2** [comment]
Access to 80 and 443 is working, but not to 30001:
```
# k -n ingress-nginx get svc
NAME                                 TYPE           CLUSTER-IP     EXTERNAL-IP     PORT(S)                                      AGE
ingress-nginx-controller             LoadBalancer   10.150.32.9    10.224.1.112    80:30345/TCP,443:32012/TCP,30001:32100/TCP   5d21h
ingress-nginx-controller-admission   ClusterIP      10.150.7.234   <none>          443/TCP                                      5d21h
```

**Comment by tgdfool2** [comment]
As suggested by @oilbeater, I tried to delete the `LoadBalancer` Service and recreate it with all the ports configured "from scratch" (80, 443 and 30001), instead of creating the Service with only 80 and 443, and then adding 30001 at a later stage. And with this setup, all the ports are properly reachable.

So this definitely looks like a bug in the way new ports are added to the `LoadBalancer` Service.

@oilbeater: please let me know if you need additional information to help you fixing the issue.

Thanks!
